File: src/test/resources/org/bau/benchmarks/swift/linkedList.swift
// https://lemire.me/blog/2026/02/15/how-bad-can-python-stop-the-world-pauses-get/

import Foundation

final class Node {
    var value: Int
    var next: Node?

    init(value: Int, next: Node? = nil) {
        self.value = value
        self.next = next
    }
}

func createLinkedList(_ len: Int) -> Node {
    let head = Node(value: 0)
    var current = head
    for i in 1..<len {
        let newNode = Node(value: i)
        current.next = newNode
        current = newNode
    }
    return head
}

print("Starting stress test...")
let x = createLinkedList(50_000_000)
let count = 1_000_000
let len = 1000
var worst: UInt64 = 0
let start = DispatchTime.now().uptimeNanoseconds
var last = start
for _ in 0..<count {
    _ = createLinkedList(len)
    let now = DispatchTime.now().uptimeNanoseconds
    let delta = now - last
    if delta > worst {
        worst = delta
    }
    last = now
}
let total = DispatchTime.now().uptimeNanoseconds - start
let avg = total / UInt64(count)
print(String(
    format: "Max delay: %.3f ms, avg %.3f ms; dummy=%d",
    Double(worst) / 1_000_000.0,
    Double(avg) / 1_000_000.0,
    x.value
))

File: src/test/resources/org/bau/benchmarks/python/linkedList.py
# https://lemire.me/blog/2026/02/15/how-bad-can-python-stop-the-world-pauses-get/

import time

class Node:
    def __init__(self, value):
        self.value = value
        self.next = None
    def add_next(self, node):
        self.next = node

def create_linked_list(limit):
    """ create a linked list of length 'limit' """
    head = Node(0)
    current = head
    for i in range(1, limit):
        new_node = Node(i)
        current.add_next(new_node)
        current = new_node
    return head

def main():
    print("Python")
    x = create_linked_list(50_000_000)
    count = 1_000_000
    len = 1000
    worst = 0.0
    last = time.perf_counter_ns()
    for i in range(1_000_000):
        create_linked_list(1000)
        now = time.perf_counter_ns()
        worst = max(worst, now - last)
        last = now
    print(f"Max delay: {worst / 1_000_000} ms; dummy value: {x.value}", flush=True)

main()